src/analysis/nextest/nextest_line_analyser.rs

```diff
@@ -48,31 +53,68 @@ impl LineAnalyzer for NextestLineAnalyzer {
         self.default_analyzer.analyze_line(cmd_line)
     }
 }
-
-fn title_key(content: &TLine) -> Option<String> {
-    title_key_v1(content).or_else(|| title_key_v2(content))
-}
-/// Return the key when the line is like "--- STD(OUT|ERR): somekey ---"
-fn title_key_v1(content: &TLine) -> Option<String> {
-    let mut strings = content.strings.iter();
-    let first = strings.next()?;
-    if !regex_is_match!(r"^--- STD(OUT|ERR):\s*", &first.raw) {
-        return None;
+impl NextestLineAnalyzer {
+    fn stdx_section_key(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        self.stdx_section_key_v1(content)
+            .or_else(|| self.stdx_section_key_v2(content))
+            .or_else(|| self.stdx_section_key_v3(content))
     }
-    extract_key_after_crate_name_v1(strings)
-}
-/// Return the key when the line is like "──── STD(OUT|ERR): cratename some::key"
-fn title_key_v2(content: &TLine) -> Option<String> {
-    let mut strings = content.strings.iter();
-    let ts = strings.next()?;
-    if ts.csi != CSI_ERROR || ts.raw != "────" {
-        return None;
+    /// Return the key when the line is like "--- STD(OUT|ERR): somekey ---"
+    fn stdx_section_key_v1(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        let mut strings = content.strings.iter();
+        let first = strings.next()?;
+        if !regex_is_match!(r"^--- STD(OUT|ERR):\s*", &first.raw) {
+            return None;
+        }
+        extract_key_after_crate_name_v1(strings)
     }
-    let ts = strings.find(|ts| ts.raw != " ")?;
-    if !regex_is_match!(r"^STD(OUT|ERR):\s*", &ts.raw) {
-        return None;
+    /// Return the key when the line is like "──── STD(OUT|ERR): cratename some::key"
+    fn stdx_section_key_v2(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        let mut strings = content.strings.iter();
+        let ts = strings.next()?;
+        if ts.csi != CSI_ERROR || ts.raw != "────" {
+            return None;
+        }
+        let ts = strings.find(|ts| ts.raw != " ")?;
+        if !regex_is_match!(r"^STD(OUT|ERR):\s*", &ts.raw) {
+            return None;
+        }
+        extract_key_after_crate_name_v2(strings)
+    }
+    /// Return the last test key when the line is like "──── std(out|err)"
+    /// For nextest 0.9.95+ (see https://github.com/Canop/bacon/issues/350)
+    ///
+    fn stdx_section_key_v3(
+        &self,
+        content: &TLine,
+    ) -> Option<String> {
+        let mut strings = content.strings.iter().skip_while(|ts| ts.is_blank());
+        let ts = strings.next()?;
+        if ts.csi != CSI_ERROR || (ts.raw != "stderr" && ts.raw != "stdout") {
+            return None;
+        }
```

Comment:
While you're here I'd also support `output`, which is for situations where stdout and stderr are combined. This is currently semi-hidden but might become more visible soon.

Reply:
Added "output"